yarn-project/pxe/src/kernel_prover/bb_prover/bb_native_proof_creator.ts

```diff
@@ -0,0 +1,713 @@
+import {
+  Fr,
+  type PrivateCircuitPublicInputs,
+  type PrivateKernelCircuitPublicInputs,
+  type PrivateKernelInitCircuitPrivateInputs,
+  type PrivateKernelInnerCircuitPrivateInputs,
+  type PrivateKernelTailCircuitPrivateInputs,
+  type PrivateKernelTailCircuitPublicInputs,
+  Proof,
+  type VERIFICATION_KEY_LENGTH_IN_FIELDS,
+  makeEmptyProof,
+} from '@aztec/circuits.js';
+import { siloNoteHash } from '@aztec/circuits.js/hash';
+import { randomBytes, sha256 } from '@aztec/foundation/crypto';
+import { type LogFn, createDebugLogger } from '@aztec/foundation/log';
+import { type Tuple } from '@aztec/foundation/serialize';
+import { Timer } from '@aztec/foundation/timer';
+import {
+  ClientCircuitArtifacts,
+  type ClientProtocolArtifact,
+  convertPrivateKernelInitInputsToWitnessMap,
+  convertPrivateKernelInitOutputsFromWitnessMap,
+  convertPrivateKernelInnerInputsToWitnessMap,
+  convertPrivateKernelInnerOutputsFromWitnessMap,
+  convertPrivateKernelTailForPublicOutputsFromWitnessMap,
+  convertPrivateKernelTailOutputsFromWitnessMap,
+  executeTail,
+  executeTailForPublic,
+} from '@aztec/noir-protocol-circuits-types';
+import { type ACVMField, WASMSimulator } from '@aztec/simulator';
+import { type NoirCompiledCircuit } from '@aztec/types/noir';
+
+import { type WitnessMap } from '@noir-lang/acvm_js';
+import { serializeWitness } from '@noir-lang/noirc_abi';
+import * as proc from 'child_process';
+import * as fs from 'fs/promises';
+
+import { type ProofCreator, type ProofOutput } from '../interface/proof_creator.js';
+
+/**
+ * Temporary implementation of ProofCreator using the native bb binary.
+ * Will be replaced by the WASM equivalent once ready
+ */
+
+const VK_FILENAME = 'vk';
+const VK_FIELDS_FILENAME = 'vk_fields.json';
+const PROOF_FILENAME = 'proof';
+//const PROOF_FIELDS_FILENAME = 'proof_fields.json';
+
+//const AGGREGATION_OBJECT_SIZE = 16;
+const CIRCUIT_SIZE_INDEX = 3;
+const CIRCUIT_PUBLIC_INPUTS_INDEX = 4;
+const CIRCUIT_RECURSIVE_INDEX = 5;
+
+enum BB_RESULT {
+  SUCCESS,
+  FAILURE,
+  ALREADY_PRESENT,
+}
+
+type BBSuccess = {
+  status: BB_RESULT.SUCCESS | BB_RESULT.ALREADY_PRESENT;
+  duration: number;
+  pkPath?: string;
+  vkPath?: string;
+  proofPath?: string;
+};
+
+type BBFailure = {
+  status: BB_RESULT.FAILURE;
+  reason: string;
+};
+
+/**
+ * Invokes the Barretenberg binary with the provided command and args
+ * @param pathToBB - The path to the BB binary
+ * @param command - The command to execute
+ * @param args - The arguments to pass
+ * @param logger - A log function
+ * @param resultParser - An optional handler for detecting success or failure
+ * @returns The completed partial witness outputted from the circuit
+ */
+function executeBB(
```

Comment:
I was going to say "it would be nice to have a wrapper to the bb binary" but then I remembered these proofs should actually be generated through wasm